bitnami/thanos/templates/query/deployment.yaml

```diff
@@ -80,10 +80,13 @@ spec:
             - --store=dnssrv+_grpc._tcp.{{- include "common.tplvalues.render" ( dict "value" $query.dnsDiscovery.sidecarsService "context" $) -}}.{{- include "common.tplvalues.render"  ( dict "value" $query.dnsDiscovery.sidecarsNamespace "context" $) -}}.svc.{{ .Values.clusterDomain }}
             {{- end }}
             {{- if and .Values.storegateway.enabled $query.dnsDiscovery.enabled }}
-            - --store=dnssrv+_grpc._tcp.{{ include "common.names.fullname" . }}-storegateway.{{ .Release.Namespace }}.svc.{{ .Values.clusterDomain }}
+            - --store=dnssrv+_grpc._tcp.{{ include "common.names.fullname" . }}-storegateway-headless.{{ .Release.Namespace }}.svc.{{ .Values.clusterDomain }}
```

Comment:
The headless service can be disabled. What happens if the headless service is disabled?

Reply:
hmm we could add it as a flag.
headless service just makes more sense.

Reply:
like that?